frontend/docsettings.lua

```diff
@@ -18,20 +18,34 @@ local HISTORY_DIR = DataStorage:getHistoryDir()
 
 local function buildCandidates(list)
     local candidates = {}
+    local previous_entry_exists = false
 
     for i, file_path in ipairs(list) do
         -- Ignore missing files.
         if file_path ~= "" and lfs.attributes(file_path, "mode") == "file" then
+            local mtime = lfs.attributes(file_path, "modification")
+            -- NOTE: Extra trickery: if we're inserting a "backup" file, and its main buddy exists,
+            --       make sure it will *never* sort ahead of it by negating its mtime.
+            --       This aims to avoid weird UTC/localtime issues when USBMS is involved,
+            --       c.f., https://github.com/koreader/koreader/issues/9227#issuecomment-1345263324
+            if file_path:sub(-4) == ".old" and previous_entry_exists then
+                mtime = -mtime
```

Comment:
It's possible the main buddy exists, but will be empty/corrupted/invalid, while the old would be ok.
But if there's a legacy docsetting, it will have a positive timestamp, and will be used instead of the .old :/

Reply:
May be just remember `previous_entry_mtime` and use `mtime = previous_entry_mtime - 1` for the  .old ?

Reply:
Good catch. I'll use the main's mtime instead, then.

(I wanted a way to keep the spirit of the real mtime alive somehow, but you just shot a nice round hole in that approach ^^).

Reply:
Done (with the backup's mtime, actually, so that we always use the most recent timestamp); and added logging to make the switcheroo explicit (and only as necessary).

```
2/12/22-02:10:12 WARN  DocSettings: Backup /var/tmp/niluje/koreader/test/juliet.sdr/metadata.epub.lua.old is newer ( 1670804971 ) than its primary ( 1670804923 ), fudging timestamps! 
12/12/22-02:10:12 DEBUG DocSettings: data is read from /var/tmp/niluje/koreader/test/juliet.sdr/metadata.epub.lua
```